src/pages/waiting-room-to-car/cat-be/components/vehicle-checks-modal/__tests__/vehicle-checks-modal.cat-be.analytics.effects.spec.ts

```diff
@@ -0,0 +1,135 @@
+import { TestBed } from '@angular/core/testing';
+import { VehicleChecksModalAnalyticsEffects } from '../vehicle-checks-modal.cat-be.analytics.effects';
+import { Store, StoreModule } from '@ngrx/store';
+import { StoreModel } from '../../../../../../shared/models/store.model';
+import { ReplaySubject } from 'rxjs/ReplaySubject';
+import { testsReducer } from '../../../../../../modules/tests/tests.reducer';
+import { AnalyticsProvider } from '../../../../../../providers/analytics/analytics';
+import { AnalyticsProviderMock } from '../../../../../../providers/analytics/__mocks__/analytics.mock';
+import { provideMockActions } from '@ngrx/effects/testing';
+import * as testsActions from '../../../../../../modules/tests/tests.actions';
+import { TestCategory } from '../../../../../../shared/models/test-category';
+import { VehicleChecksViewDidEnter } from '../vehicle-checks-modal.cat-be.actions';
+import { AnalyticRecorded } from '../../../../../../providers/analytics/analytics.actions';
+import { AnalyticsEventCategories, AnalyticsScreenNames } from '../../../../../../providers/analytics/analytics.model';
+import * as VehicleChecksActions
+  from '../../../../../../modules/tests/test-data/vehicle-checks/vehicle-checks.cat-be.action';
+
+import {
+  QuestionOutcome,
+  QuestionResult,
+} from '@dvsa/mes-test-schema/categories/common';
+
+describe('Vehicle Checks Modal Analytics Effects', () => {
+
+  let effects: VehicleChecksModalAnalyticsEffects;
+  let analyticsProviderMock;
+  let actions$: any;
+  let store$: Store<StoreModel>;
+  const screenName = AnalyticsScreenNames.VEHICLE_CHECKS;
+
+  beforeEach(() => {
+    actions$ = new ReplaySubject(1);
+    TestBed.configureTestingModule({
+      imports: [
+        StoreModule.forRoot({
+          tests: testsReducer,
+        }),
+      ],
+      providers: [
+        VehicleChecksModalAnalyticsEffects,
+        { provide: AnalyticsProvider, useClass: AnalyticsProviderMock },
+        provideMockActions(() => actions$),
+        Store,
+      ],
+    });
+    effects = TestBed.get(VehicleChecksModalAnalyticsEffects);
+    analyticsProviderMock = TestBed.get(AnalyticsProvider);
+    store$ = TestBed.get(Store);
+  });
+
+  describe('vehicleChecksModalViewDidEnter$ effect', () => {
+    it('should call analytics.setCurrentPage', (done) => {
+      store$.dispatch(new testsActions.StartTest(12345, TestCategory.BE));
+      actions$.next(new VehicleChecksViewDidEnter());
+      effects.vehicleChecksModalViewDidEnter$.subscribe((result) => {
+        expect(result instanceof AnalyticRecorded).toBe(true);
+        expect(analyticsProviderMock.setCurrentPage).toHaveBeenCalledWith(screenName);
+        done();
+      });
+    });
+  });
+
+  describe('showMeQuestionChanged$ effect', () => {
+    const questionNumber: number = 1;
+    const showMeQuestion: QuestionResult = {
+      code: 'S01',
+    };
+    it('should log an analyics event with show me question info', () => {
```

Comment:
When doing assertions in a `.subscribe()` callback you need to call `done()`, otherwise your assertions will never execute and tests will pass regardless.

Reply:
updated, thanks for spotting!